src/httpauth/basic.c

```diff
@@ -123,3 +123,120 @@ int httpauth_basic_encode(const struct httpauth_basic *basic, struct mbuf *mb)
 	mbuf_set_pos(mb, 0);
 	return 0;
 }
+
+/* HTTPAUTH BASIC REQUESTS*/
+
+static void httpauth_basic_request_destructor(void *arg)
+{
+	struct httpauth_basic_req *req = arg;
+
+	mem_deref(req->realm);
+	mem_deref(req->charset);
+}
+
+
+int httpauth_basic_request_print(struct re_printf *pf,
+	const struct httpauth_basic_req *req)
+{
+	if (!req)
+		return 0;
+
+	if (req->charset && str_len(req->charset))
+		return re_hprintf(pf, "Basic realm=\"%s\", charset=\"%s\"",
+			req->realm, req->charset);
+	else
+		return re_hprintf(pf, "Basic realm=\"%s\"", req->realm);
```

Comment:
this code can be simplified.

Print "basic realm" every time. If charset is set, append charset.